learning/tour-of-beam/frontend/lib/pages/tour/widgets/unit_content.dart

```diff
@@ -17,66 +17,212 @@
  */
 
 import 'package:flutter/material.dart';
-import 'package:flutter_markdown/flutter_markdown.dart';
-import 'package:markdown/markdown.dart' as md;
 import 'package:playground_components/playground_components.dart';
 
+import '../../../constants/sizes.dart';
 import '../../../models/unit_content.dart';
+import '../state.dart';
+import 'complete_unit_button.dart';
+import 'hints.dart';
+import 'markdown/tob_markdown.dart';
+import 'solution_button.dart';
 
 class UnitContentWidget extends StatelessWidget {
+  final TourNotifier tourNotifier;
+
+  const UnitContentWidget(this.tourNotifier);
+
+  @override
+  Widget build(BuildContext context) {
+    final themeData = Theme.of(context);
+
+    return Container(
+      height: MediaQuery.of(context).size.height -
+          BeamSizes.appBarHeight -
+          TobSizes.footerHeight,
```

Comment:
Can this be auto-calculated by Flutter somehow? Or is this for narrow screens where Playground is below the unit content? Anyway it feels like it is not the content widget's job to query the screen for its height. This should be done by some higher-level layout widget, and the unit content widget should just be given a height to fill.